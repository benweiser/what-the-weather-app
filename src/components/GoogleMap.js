import React from 'react';

import GoogleMapReact from 'google-map-react';

const EventMarker = ({ children }) => (
  <div style={{ color: 'white', fontSize: '21px' }}>{children}</div>
);

const Map = props => {
  const { center } = props;

  return (
    <div style={{ height: '400px', width: '100%' }}>
      <GoogleMapReact
        bootstrapURLKeys={{ key: 'AIzaSyCjNFYH29kyE1Vwl-b3VvlrtBsQ8nV995A' }}
        defaultCenter={center}
        defaultZoom={props.zoom}
        options={{
          styles: [
            {
              elementType: 'geometry',
              stylers: [{ color: '#242f3e' }]
            },
            {
              elementType: 'labels.text.stroke',
              stylers: [{ color: '#242f3e' }]
            },
            {
              elementType: 'labels.text.fill',
              stylers: [{ color: '#746855' }]
            },
            {
              featureType: 'administrative.locality',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#d59563' }]
            },
            {
              featureType: 'poi',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#f5f5eb' }]
            },
            {
              featureType: 'poi.park',
              elementType: 'geometry',
              stylers: [{ color: '#263c3f' }]
            },
            {
              featureType: 'poi.park',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#6b9a76' }]
            },
            {
              featureType: 'road',
              elementType: 'geometry',
              stylers: [{ color: '#38414e' }]
            },
            {
              featureType: 'road',
              elementType: 'geometry.stroke',
              stylers: [{ color: '#212a37' }]
            },
            {
              featureType: 'road',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#9ca5b3' }]
            },
            {
              featureType: 'road.highway',
              elementType: 'geometry',
              stylers: [{ color: '#746855' }]
            },
            {
              featureType: 'road.highway',
              elementType: 'geometry.stroke',
              stylers: [{ color: '#1f2835' }]
            },
            {
              featureType: 'road.highway',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#f3d19c' }]
            },
            {
              featureType: 'transit',
              elementType: 'geometry',
              stylers: [{ color: '#2f3948' }]
            },
            {
              featureType: 'transit.station',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#d59563' }]
            },
            {
              featureType: 'water',
              elementType: 'geometry',
              stylers: [{ color: '#17263c' }]
            },
            {
              featureType: 'water',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#515c6d' }]
            },
            {
              featureType: 'water',
              elementType: 'labels.text.stroke',
              stylers: [{ color: '#17263c' }]
            }
          ]
        }}
      >
        <EventMarker
          marker={444}
          key={4}
          lat={center.lat}
          lng={center.lng}
          showBalloon
        >
          <i className="fa fa-map-marker" aria-hidden="true" />
        </EventMarker>
      </GoogleMapReact>
    </div>
  );
};

Map.defaultProps = {
  zoom: 12
};

export default Map;
