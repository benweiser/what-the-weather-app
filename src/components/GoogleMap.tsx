import React, { memo } from 'react';

import GoogleMapReact, { ChangeEventValue } from 'google-map-react';
import { googleMapAPIKey } from '../apiKey';

export interface MapProps extends Partial<ChangeEventValue> {}

export interface EventMarkerInterface {
  marker?: number;
  key?: number;
  lat: number;
  lng: number;
  showBalloon?: boolean;
  children?: React.ReactChild;
}

const EventMarker = ({ children }: EventMarkerInterface) => (
  <div style={{ color: 'white', fontSize: '21px' }}>{children}</div>
);

const Map = (props: MapProps) => {
  const { center } = props;

  return (
    <div style={{ height: '400px', width: '100%' }}>
      <GoogleMapReact
        bootstrapURLKeys={{ key: googleMapAPIKey }}
        defaultCenter={center}
        defaultZoom={props.zoom}
        options={{
          styles: [
            {
              elementType: 'geometry',
              stylers: [{ color: '#242f3e' }]
            },
            {
              elementType: 'labels.text.stroke',
              stylers: [{ color: '#242f3e' }]
            },
            {
              elementType: 'labels.text.fill',
              stylers: [{ color: '#746855' }]
            },
            {
              featureType: 'administrative.locality',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#d59563' }]
            },
            {
              featureType: 'poi',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#f5f5eb' }]
            },
            {
              featureType: 'poi.park',
              elementType: 'geometry',
              stylers: [{ color: '#263c3f' }]
            },
            {
              featureType: 'poi.park',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#6b9a76' }]
            },
            {
              featureType: 'road',
              elementType: 'geometry',
              stylers: [{ color: '#38414e' }]
            },
            {
              featureType: 'road',
              elementType: 'geometry.stroke',
              stylers: [{ color: '#212a37' }]
            },
            {
              featureType: 'road',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#9ca5b3' }]
            },
            {
              featureType: 'road.highway',
              elementType: 'geometry',
              stylers: [{ color: '#746855' }]
            },
            {
              featureType: 'road.highway',
              elementType: 'geometry.stroke',
              stylers: [{ color: '#1f2835' }]
            },
            {
              featureType: 'road.highway',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#f3d19c' }]
            },
            {
              featureType: 'transit',
              elementType: 'geometry',
              stylers: [{ color: '#2f3948' }]
            },
            {
              featureType: 'transit.station',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#d59563' }]
            },
            {
              featureType: 'water',
              elementType: 'geometry',
              stylers: [{ color: '#17263c' }]
            },
            {
              featureType: 'water',
              elementType: 'labels.text.fill',
              stylers: [{ color: '#515c6d' }]
            },
            {
              featureType: 'water',
              elementType: 'labels.text.stroke',
              stylers: [{ color: '#17263c' }]
            }
          ]
        }}
      >
        {center && (
          <EventMarker
            marker={444}
            key={4}
            lat={center.lat}
            lng={center.lng}
            showBalloon
          >
            <i className="fa fa-map-marker" aria-hidden="true" />
          </EventMarker>
        )}
      </GoogleMapReact>
    </div>
  );
};

Map.defaultProps = {
  zoom: 12
};

export default memo(Map);
