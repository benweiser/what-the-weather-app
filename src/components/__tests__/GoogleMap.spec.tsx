import { render } from 'react-testing-library';
import React from 'react';
import GoogleMap from '../GoogleMap';

describe('Google Map Component', () => {
  it('should render a Google Map', () => {
    const { container } = render(
      <GoogleMap
        center={{
          lat: 323,
          lng: 323
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
