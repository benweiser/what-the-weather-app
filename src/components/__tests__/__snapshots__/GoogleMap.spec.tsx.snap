// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Map Component should render a Google Map 1`] = `
<div>
  <div
    style="height: 400px; width: 100%;"
  >
    <div
      style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
    >
      <div
        style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
      />
      <div
        style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
      >
        <div
          style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
        >
          <div
            style="width: 0px; height: 0px; left: 0px; top: 0px; background-color: transparent; position: absolute;"
          >
            <div
              style="color: white; font-size: 21px;"
            >
              <i
                aria-hidden="true"
                class="fa fa-map-marker"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
